src/client/pages/SignIn.tsx

```diff
@@ -31,77 +35,70 @@ const Links = ({ children }: { children: React.ReactNode }) => (
   </p>
 );
 
-export const SignIn = () => {
-  const clientState: ClientState = useContext(ClientStateContext);
-  const { pageData = {} } = clientState;
-  const { returnUrl } = pageData;
-  const returnUrlQuery = returnUrl
-    ? `?returnUrl=${encodeURIComponent(returnUrl)}`
-    : '';
-
-  return (
-    <>
-      <Header />
-      <Nav
-        tabs={[
-          {
-            displayText: PageTitle.SIGN_IN,
-            linkTo: Routes.SIGN_IN,
-            isActive: true,
-          },
-          {
-            displayText: PageTitle.REGISTRATION,
-            linkTo: Routes.REGISTRATION,
-            isActive: false,
-          },
-        ]}
-      />
-      <Main>
-        <PageBox>
-          <PageBody>
-            <form
-              css={form}
-              method="post"
-              action={`${Routes.SIGN_IN}${returnUrlQuery}`}
-            >
-              <CsrfFormField />
-              <TextInput
-                css={textInput}
-                label="Email"
-                name="email"
-                type="email"
-              />
-              <TextInput
-                css={textInput}
-                label="Password"
-                name="password"
-                type="password"
-              />
-              <Links>
-                <Link subdued={true} href="/reset">
-                  Reset password
-                </Link>{' '}
-                or{' '}
-                <Link subdued={true} href="/magic-link">
-                  email me a link to sign in
-                </Link>
-              </Links>
-              <Button css={button} type="submit" data-cy="sign-in-button">
-                Sign in
-              </Button>
-            </form>
-            <Divider
-              size="full"
-              spaceAbove="loose"
-              displayText="or continue with"
+export const SignIn = ({ queryString, errorSummary, email }: SignInProps) => (
+  <>
+    <Header />
+    <Nav
+      tabs={[
+        {
+          displayText: PageTitle.SIGN_IN,
+          linkTo: Routes.SIGN_IN,
+          isActive: true,
+        },
+        {
+          displayText: PageTitle.REGISTRATION,
+          linkTo: Routes.REGISTRATION,
+          isActive: false,
+        },
+      ]}
+    />
+    <Main>
+      <PageBox>
+        <PageBody>
+          <form
+            css={form}
+            method="post"
+            action={`${Routes.SIGN_IN}${queryString}`}
```

Comment:
We could use `joinUrl` here? Although it's kind of overkill because we control both strings, what do you think?

Reply:
Would joinUrl work in this case? As I'm appending the `queryString` parameters instead rather than a url.

I think there may be a case for a lib method to append query parameter strings to urls though, e.g. checks if a `?` exists in the url or not, and appends the query parameters as appropriate?

Reply:
Ah yes, you're right, it won't work here! You could maybe use `URL`? Might be overkill though

Reply:
Again after so long...

We refactored completely how we do query params in this PR: https://github.com/guardian/gateway/pull/1109